plugins/modules/zpool.py

```diff
@@ -0,0 +1,588 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2025, Tom Hesse <contact@tomhesse.xyz>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+module: zpool
+short_description: Manage ZFS zpools
+description:
+  - Create, destroy, and modify ZFS zpools and their vdev layouts, pool properties, and filesystem properties.
+extends_documentation_fragment:
+  - community.general.attributes
+attributes:
+  check_mode:
+    support: partial
+    details:
+      - In check_mode, any zpool subcommand that supports the dry-run flag (`-n`) will be run with `-n` and its
+        simulated output included in the module's diff results.
+  diff_mode:
+    support: full
+author:
+  - Tom Hesse (@tomhesse)
+options:
+  name:
+    description:
+      - Name of the zpool to manage.
+    required: true
+    type: str
+  state:
+    description:
+      - Whether the pool should exist.
+    choices: [ present, absent ]
+    default: present
+    type: str
+  disable_new_features:
+    description:
+      - If true, disable new ZFS feature flags when creating.
+    type: bool
+    default: false
+  force:
+    description:
+      - If true, force operations (e.g. overwrite existing devices).
+    type: bool
+    default: false
+  pool_properties:
+    description:
+      - Dictionary of ZFS pool properties to set (e.g. autoexpand, cachefile).
+    type: dict
+    default: {}
+  filesystem_properties:
+    description:
+      - Dictionary of ZFS filesystem properties to set on the root dataset (e.g. compression, dedup).
+    type: dict
+    default: {}
+  mountpoint:
+    description:
+      - Filesystem mountpoint for the root dataset.
+    type: str
+  altroot:
+    description:
+      - Alternate root for mounting filesystems.
+    type: str
+  temp_name:
+    description:
+      - Temporary name used during pool creation.
+    type: str
+  vdevs:
+    description:
+      - List of vdev definitions for the pool.
+    required: true
+    type: list
+    elements: dict
+    suboptions:
+      role:
+        description:
+          - Special vdev role (e.g. log, cache, spare).
+        type: str
+        choices: [ log, cache, spare, dedup, special ]
+      type:
+        description:
+          - Vdev topology (stripe, mirror, raidz).
+        type: str
+        choices: [ stripe, mirror, raidz, raidz1, raidz2, raidz3 ]
+        default: stripe
+      disks:
+        description:
+          - List of device paths to include in this vdev.
+        type: list
+        elements: str
+'''
+
+EXAMPLES = r'''
+- name: Create pool "tank" on /dev/sda
+  community.general.zpool:
+    name: tank
+    vdevs:
+    - disks:
+        - /dev/sda
```

Comment:
```suggestion
      - disks:
          - /dev/sda
```
Please indent YAML lists. (Same below.)